src/hl/container.rs

```diff
@@ -346,6 +347,98 @@ impl<'a> Writer<'a> {
     }
 }
 
+#[derive(Debug)]
+pub struct ByteReader<'a> {
+    obj: &'a Container,
```

Comment:
Is it easier to work with the ByteReader if it is owned? Clone is cheap as the object is refcounted

Reply:
Sure, I guess it makes sense. Then the `Container` needs to be cloned on construction, correct?

Reply:
Yes, that sounds right